src/vaults/SyncRequests.sol

```diff
@@ -60,10 +61,27 @@ contract SyncRequests is BaseInvestmentManager, ISyncRequests {
         emit File(what, data);
     }
 
+    /// --- IUpdateContract ---
+    /// @inheritdoc IUpdateContract
+    function update(uint64 poolId, bytes16 scId, bytes memory payload) external auth {
+        uint8 kind = uint8(MessageLib.updateContractType(payload));
+
+        if (kind == uint8(UpdateContractType.Valuation)) {
+            MessageLib.UpdateContractValuation memory m = MessageLib.deserializeUpdateContractValuation(payload);
+
+            require(poolManager.shareToken(poolId, scId) != address(0), "SyncRequests/share-token-does-not-exist");
+            (address asset, uint256 tokenId) = poolManager.idToAsset(m.assetId);
+
+            this.setValuation(m.poolId, m.scId, asset, tokenId, address(bytes20(m.valuation)));
```

Comment:
This should be

```suggestion
            setValuation(m.poolId, m.scId, asset, tokenId, address(bytes20(m.valuation)));
```

And `setValuation` public

Reply:
Should it really be public though? This call does not need to be exposed publicly.

Reply:
Maybe should be private! or just inlined